internal/services/containerapps/helpers/container_app_environment.go

```diff
@@ -37,6 +37,7 @@ func WorkloadProfileSchema() *pluginsdk.Schema {
 					Type:     pluginsdk.TypeString,
 					Required: true,
 					ValidateFunc: validation.StringInSlice([]string{
+						"Consumption",
```

Comment:
Based on the offical MS docs the type name for consumption should be small letters. `consumption` not `Consumption` .
https://learn.microsoft.com/en-us/azure/container-apps/workload-profiles-overview#profile-types